bento/src/main/java/com/yelp/android/bento/core/AsyncInflationBridge.kt

```diff
@@ -146,4 +150,30 @@ internal class AsyncInflationBridge @JvmOverloads constructor(
             }
         }
     }
+
+    private fun findAndAttachToLifecycleOwner() {
+        var owner = recyclerView.findViewTreeLifecycleOwner()
+        if (owner == null) {
+            if (recyclerView.context is LifecycleOwner) {
+                owner = recyclerView.context as LifecycleOwner
+            }
+        }
+        owner?.lifecycleScope?.launch(Dispatchers.Main) {
+            owner.lifecycle.addObserver(object : LifecycleObserver {
+
+                @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+                fun cleanUp() {
+                    clearResources()
+                    owner.lifecycle.removeObserver(this)
+                }
+            })
+        }
+    }
```

Comment:
I don't think you need to wrap the addObserver in a coroutine, can simply do

```
private fun findAndAttachToLifecycleOwner() {
    var owner = recyclerView.findViewTreeLifecycleOwner()
    if (owner == null) {
        if (recyclerView.context is LifecycleOwner) {
            owner = recyclerView.context as LifecycleOwner
        }
    }
    owner?.lifecycle?.addObserver(object : LifecycleObserver {
        @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
        fun cleanUp() {
            clearResources()
            owner.lifecycle.removeObserver(this)
        }
    })
}
```
Adding an observer is async by definition right, plus the findAndAttachToLifecycleOwner() would only be called from the main thread, as the init would happen on the main thread.

At the very least, a `Dispatchers.Main.immediate` should be used instead of the `Dispatchers.Main`?

Reply:
@redwarp If there is a RecyclerView in a ViewHolder then I think it will still be created on the background thread.

Reply:
Ah, good point! Let's go for Dispatchers.Main.immediate then!

Reply:
Sounds good!